lib/plausible/open_telemetry.ex

```diff
@@ -0,0 +1,69 @@
+defmodule Plausible.OpenTelemetry do
+  @moduledoc false
+
+  require OpenTelemetry.Tracer, as: Tracer
+
+  def add_site_attributes(%Plausible.Site{} = site) do
+    Tracer.set_attributes([
+      {"plausible.site.id", site.id},
+      {"plausible.site.domain", site.domain}
+    ])
+  end
+
+  def add_user_attributes(%Plausible.Auth.User{} = user) do
+    Tracer.set_attributes([
+      {"plausible.user.id", user.id},
+      {"plausible.user.name", user.name},
+      {"plausible.user.email", user.email}
+    ])
+  end
+
+  # https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/README.md#service
+  def resource_attributes(build_metadata) do
+    [
+      {"service.name", "analytics"},
+      {"service.namespace", "plausible"},
+      {"service.instance.id", "app00"},
+      {"service.version", build_metadata[:version]}
+    ]
+  end
+end
+
+defmodule Plausible.OpenTelemetry.Sampler do
+  @moduledoc """
+  [Custom OpenTelemetry sampler](https://hexdocs.pm/opentelemetry/readme.html#samplers)
+  implementation that samples 1% of the `/api/event` traces, but records 100% of
+  other traces.
+  """
+
+  @behaviour :otel_sampler
+  require OpenTelemetry.Tracer, as: Tracer
+
+  @ratio_sampler :otel_sampler_trace_id_ratio_based.setup(0.01)
+
+  @impl true
+  def setup(_sampler_opts), do: []
+
+  @impl true
+  def description(_sampler_config), do: "#{__MODULE__}"
```

Comment:
```suggestion
  def description(_sampler_config), do: inspect(__MODULE__)
```

unless we want to keep the `Elixir.` prefix